src/Icicle/Sea/FromAvalanche/Psv.hs

```diff
@@ -462,77 +463,159 @@ seaOfWriteProgramOutput state = do
 
 seaOfOutput :: Doc -> OutputName -> ValType -> [ValType] -> Int -> Either SeaError Doc
 seaOfOutput ps oname@(OutputName name) otype0 ts0 ixStart
-  = let members     = List.take (length ts0) (fmap (\ix -> ps <> "->" <> seaOfNameIx name ix) [ixStart..])
+  = let dprintf n   = "dprintf (fd, \"%s|" <> attrib <> "|%s\\n\"" <> ", entity, " <> n <> ");"
         attrib      = seaOfEscaped name
+        free s      = indent 4 $ "free (" <> s <> ");"
+        calloc  n s = "char *" <> n <> " = calloc(1," <> s <> " );"
+        decli   n   = "int   " <> n <> " = 0;"
+    in  do (str, bufs) <- strOfOutput ps oname otype0 ts0 ixStart
+           case bufs of
+             ((b,_,_):_) -> pure
+                    $ vsep
+                    $    ["{" ]
+                      <> fmap (\(n,s,_) -> indent 4 $ calloc n s) (List.reverse bufs)
```

Comment:
If you wanted you could allocate this out of the mempool which will avoid the function call overhead, also you don't need to worry about freeing it then, the gc will do it.